index.d.ts

```diff
@@ -0,0 +1,50 @@
+interface ModifierArgs {
+  positional: unknown[];
+  named: { [key: string]: unknown };
+}
+
+interface IModifier<Args extends ModifierArgs = ModifierArgs> {
```

Comment:
Does something like

```
interface ScrollPositionModifierArgs {
  positional: number[],
  named: {
    delay: number
  }
}
```

...really `extends ModifierArgs`? Is that supposed to type check? cc @chriskrycho

Reply:
Yes, that’s correct. It’s a narrower type, and narrower types are always assignable where a wider type is requested. This is why you can pass a `string` where some function allows `string | number`, for example. (TS types are best understood in terms of sets and set theory!)